exporters/trace/stackdriver/src/main/java/io/opencensus/exporter/trace/stackdriver/StackdriverTraceExporter.java

```diff
@@ -52,71 +56,32 @@
-      registerInternal(StackdriverV2ExporterHandler.createWithCredentials(credentials, projectId));
-    }
-  }
-
-  /**
-   * Creates and registers the Stackdriver Trace exporter to the OpenCensus library for an explicit
-   * project ID. Only one Stackdriver exporter can be registered at any point.
-   *
-   * <p>This uses the default application credentials see {@link
-   * GoogleCredentials#getApplicationDefault}.
-   *
-   * <p>This is equivalent with:
-   *
-   * <pre>{@code
-   * StackdriverTraceExporter.createAndRegisterWithCredentialsAndProjectId(
-   *     GoogleCredentials.getApplicationDefault(), projectId);
-   * }</pre>
-   *
-   * @param projectId the cloud project id.
-   * @throws IllegalStateException if a Stackdriver exporter is already registered.
-   * @since 0.12
-   */
-  public static void createAndRegisterWithProjectId(String projectId) throws IOException {
-    synchronized (monitor) {
-      checkState(handler == null, "Stackdriver exporter is already registered.");
-      registerInternal(StackdriverV2ExporterHandler.create(projectId));
-    }
-  }
-
   /**
    * Creates and registers the Stackdriver Trace exporter to the OpenCensus library. Only one
    * Stackdriver exporter can be registered at any point.
    *
-   * <p>This uses the default application credentials see {@link
+   * <p>If the {@code credentials} in the provided {@link StackdriverTraceConfiguration} is not set,
+   * the exporter will use the default application credentials. See {@link
    * GoogleCredentials#getApplicationDefault}.
    *
-   * <p>This uses the default project ID configured see {@link ServiceOptions#getDefaultProjectId}.
-   *
-   * <p>This is equivalent with:
-   *
-   * <pre>{@code
-   * StackdriverTraceExporter.createAndRegisterWithProjectId(ServiceOptions.getDefaultProjectId());
-   * }</pre>
+   * <p>If the {@code projectId} in the provided {@link StackdriverTraceConfiguration} is not set,
+   * the exporter will use the default project ID. See {@link ServiceOptions#getDefaultProjectId}.
    *
+   * @param configuration the {@code StackdriverTraceConfiguration} used to create the exporter.
    * @throws IllegalStateException if a Stackdriver exporter is already registered.
    * @since 0.12
    */
-  public static void createAndRegister() throws IOException {
+  public static void createAndRegister(StackdriverTraceConfiguration configuration)
+      throws IOException {
     synchronized (monitor) {
       checkState(handler == null, "Stackdriver exporter is already registered.");
-      registerInternal(StackdriverV2ExporterHandler.create(ServiceOptions.getDefaultProjectId()));
+      Credentials credentials =
+          MoreObjects.<Credentials>firstNonNull(
+              configuration.getCredentials(), GoogleCredentials.getApplicationDefault());
+      String projectId =
+          MoreObjects.<String>firstNonNull(
+              configuration.getProjectId(), ServiceOptions.getDefaultProjectId());
```

Comment:
It might be better to only call GoogleCredentials.getApplicationDefault() or ServiceOptions.getDefaultProjectId() when they are needed.  For example, the Javadoc for getApplicationDefault() says that it can throw an IOException in some environments.

Reply:
Done. PTAL.